uqcsbot/scripts/events.py

```diff
@@ -104,27 +109,33 @@ def encode_text(cls, text: str) -> str:
         return text.replace("&", "&amp;").replace("<", "&lt;").replace(">", "&gt;")
 
     @classmethod
-    def from_cal_event(cls, cal_event, source: str = "UQCS"):
-        start = cal_event.get('dtstart').dt
-        end = cal_event.get('dtend').dt
-        # ical 'dt' properties are parsed as a 'DDD' (datetime, date, duration) type.
-        # The below code converts a date to a datetime, where time is set to midnight.
-        if isinstance(start, date) and not isinstance(start, datetime):
-            start = datetime.combine(start, datetime.min.time()).astimezone(utc)
-        if isinstance(end, date) and not isinstance(end, datetime):
-            end = datetime.combine(end, datetime.max.time()).astimezone(utc)
+    def from_cal_event(cls, cal_event, source: str = "UQCS", reccurance_dt=None):
```

Comment:
Could be worth adding a docstring here explaining the parameters, in particular reccurence_dt.